File: backend/internal/models/workspace.go
package models

type WorkspaceModel struct {
	ID        string `json:"id"`
	WindowID  string `json:"window_id"`
	Name      string `json:"name"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

type WorkspaceTabModel struct {
	TabID      string `json:"tab_id"`
	SessionID  string `json:"session_id"`
	WindowID   string `json:"window_id"`
	IsLocal    bool   `json:"is_local"`
	CreatedAt  string `json:"created_at"`
	ModifiedAt string `json:"modified_at"`
}

type WorkspaceRegisterWindowRequest struct {
	WindowID string `json:"window_id"`
	Mode     string `json:"mode"`
}

type WorkspaceRemoveWindowRequest struct {
	WindowID string `json:"window_id"`
}

type WorkspaceCreateRequest struct {
	WindowID string `json:"window_id"`
	Name     string `json:"name"`
}

type WorkspaceGetByWindowRequest struct {
	WindowID string `json:"window_id"`
}

type WorkspaceRegisterTabRequest struct {
	TabID     string `json:"tab_id"`
	SessionID string `json:"session_id"`
	WindowID  string `json:"window_id"`
	IsLocal   bool   `json:"is_local"`
}

type WorkspaceRemoveTabRequest struct {
	TabID string `json:"tab_id"`
}

type WorkspaceListTabsRequest struct {
	WindowID string `json:"window_id"`
}

type WorkspaceMoveTabRequest struct {
	TabID         string `json:"tab_id"`
	SourceWindow  string `json:"source_window"`
	TargetWindow  string `json:"target_window"`
	TransactionID string `json:"transaction_id,omitempty"`
}

type WorkspaceRegisterWindowResponse struct {
	Status string `json:"status"`
}

type WorkspaceRemoveWindowResponse struct {
	Status string `json:"status"`
}

type WorkspaceCreateResponse struct {
	Workspace WorkspaceModel `json:"workspace"`
}

type WorkspaceGetByWindowResponse struct {
	Workspace WorkspaceModel `json:"workspace"`
}

type WorkspaceRegisterTabResponse struct {
	Status string `json:"status"`
}

type WorkspaceRemoveTabResponse struct {
	Status string `json:"status"`
}

type WorkspaceListTabsResponse struct {
	Tabs []WorkspaceTabModel `json:"tabs"`
}

type WorkspaceMoveTabResponse struct {
	TransactionID string `json:"transaction_id"`
	TabID         string `json:"tab_id"`
	SourceWindow  string `json:"source_window"`
	TargetWindow  string `json:"target_window"`
	CompletedAt   string `json:"completed_at"`
}

type WorkspaceStateSaveRequest struct {
	ClientState map[string]interface{} `json:"client_state,omitempty"`
}

type WorkspaceStateLoadRequest struct{}

type WorkspaceStateClearRequest struct{}

type WorkspaceStateSnapshotModel struct {
	Workspaces    map[string]WorkspaceModel    `json:"workspaces"`
	WindowToSpace map[string]string            `json:"window_to_space"`
	Tabs          map[string]WorkspaceTabModel `json:"tabs"`
	WindowMode    map[string]string            `json:"window_mode"`
}

type WorkspaceStateModel struct {
	Version     int                         `json:"version"`
	SavedAt     string                      `json:"saved_at"`
	Snapshot    WorkspaceStateSnapshotModel `json:"snapshot"`
	ClientState map[string]interface{}      `json:"client_state,omitempty"`
}

type WorkspaceStateSaveResponse struct {
	Status string              `json:"status"`
	State  WorkspaceStateModel `json:"state"`
}

type WorkspaceStateLoadResponse struct {
	Found bool                 `json:"found"`
	State *WorkspaceStateModel `json:"state,omitempty"`
}

type WorkspaceStateClearResponse struct {
	Status string `json:"status"`
}

File: src/renderer/src/types/workspace.ts
export type WorkspaceWindowMode = 'primary' | 'workspace'

export interface WorkspaceModel {
  id: string
  window_id: string
  name: string
  created_at: string
  updated_at: string
}

export interface WorkspaceTabModel {
  tab_id: string
  session_id: string
  window_id: string
  is_local: boolean
  created_at: string
  modified_at: string
}

export interface WorkspaceRegisterWindowRequest {
  window_id: string
  mode: WorkspaceWindowMode
}

export interface WorkspaceRemoveWindowRequest {
  window_id: string
}

export interface WorkspaceCreateRequest {
  window_id: string
  name: string
}

export interface WorkspaceGetByWindowRequest {
  window_id: string
}

export interface WorkspaceRegisterTabRequest {
  tab_id: string
  session_id: string
  window_id: string
  is_local: boolean
}

export interface WorkspaceRemoveTabRequest {
  tab_id: string
}

export interface WorkspaceListTabsRequest {
  window_id: string
}

export interface WorkspaceMoveTabRequest {
  tab_id: string
  source_window: string
  target_window: string
  transaction_id?: string
}

export interface WorkspaceCreateResponse {
  workspace: WorkspaceModel
}

export interface WorkspaceGetByWindowResponse {
  workspace: WorkspaceModel
}

export interface WorkspaceListTabsResponse {
  tabs: WorkspaceTabModel[]
}

export interface WorkspaceMoveTabResponse {
  transaction_id: string
  tab_id: string
  source_window: string
  target_window: string
  completed_at: string
}

export interface WorkspaceStatusResponse {
  status: string
}

export interface WorkspaceStateSaveRequest {
  client_state?: Record<string, unknown>
}

export interface WorkspaceStateLoadRequest {}

export interface WorkspaceStateClearRequest {}

export interface WorkspaceStateSnapshotModel {
  workspaces: Record<string, WorkspaceModel>
  window_to_space: Record<string, string>
  tabs: Record<string, WorkspaceTabModel>
  window_mode: Record<string, string>
}

export interface WorkspaceStateModel {
  version: number
  saved_at: string
  snapshot: WorkspaceStateSnapshotModel
  client_state?: Record<string, unknown>
}

export interface WorkspaceStateSaveResponse {
  status: string
  state: WorkspaceStateModel
}

export interface WorkspaceStateLoadResponse {
  found: boolean
  state?: WorkspaceStateModel
}

export interface WorkspaceStateClearResponse {
  status: string
}
